miniwdl-plugins/s3upload/miniwdl_s3upload.py

```diff
@@ -80,6 +81,41 @@ def flag_temporary(s3uri):
     )
 
 
+def remove_temporary_flag(s3uri, retry=0):
+    """ Remove temporary flag from s3 if in outputs.json """
+    uri = urlparse(s3uri)
+    bucket, key = uri.hostname, uri.path[1:]
+    tags = s3_client.get_object_tagging(
+        Bucket=bucket,
+        Key=key,
+    )
+    remaining_tags = []
+    for tag in tags["TagSet"]:
+        if not (tag["Key"] == "swipe_temporary" and tag["Value"] == "true"):
+            remaining_tags.append(tag)
+    try:
+        if remaining_tags:
+            s3_client.put_object_tagging(
+                Bucket=bucket,
+                Key=key,
+                Tagging={
+                    'TagSet': remaining_tags
+                },
+            )
+        else:
+            s3_client.delete_object_tagging(
+                Bucket=bucket,
+                Key=key,
+            )
+    except botocore.exceptions.ClientError as e:
+        if retry > 3:
+            raise e
+        print(f"Error deleting tags for object {key} in bucket {bucket}: {e}")
+        print("Retrying in 20 seconds...")
+        time.sleep(20)
```

Comment:
Can you add a little jitter to this? Both processes may wait the 20 seconds then try again ad the exact same time.